src/gucs.rs

```diff
@@ -7,13 +7,29 @@ All rights reserved.
 Use of this source code is governed by the PostgreSQL license that can be found in the LICENSE.md file.
 */
 
+use once_cell::sync::Lazy;
 use pgx::guc::{GucContext, GucRegistry, GucSetting};
 use std::path::PathBuf;
 use std::str::FromStr;
 
 static PLRUST_WORK_DIR: GucSetting<Option<&'static str>> = GucSetting::new(None);
 static PLRUST_PG_CONFIG: GucSetting<Option<&'static str>> = GucSetting::new(None);
 static PLRUST_TRACING_LEVEL: GucSetting<Option<&'static str>> = GucSetting::new(None);
+static PLRUST_ALLOWED_DEPENDENCIES: GucSetting<Option<&'static str>> = GucSetting::new(None);
+
+static PLRUST_ALLOWED_DEPENDENCIES_CONTENTS: Lazy<toml::value::Table> = Lazy::new(|| {
+    let path = PathBuf::from_str(
+        &PLRUST_ALLOWED_DEPENDENCIES
+            .get()
+            .expect("plrust.allowed_dependencies is not set in postgresql.conf"),
+    )
+    .expect("plrust.allowed_dependencies is not a valid path");
+
+    let contents =
+        std::fs::read_to_string(&path).expect("Unable to read allow listed dependencies");
+
+    toml::from_str(&contents).expect("Unable to format allow listed dependencies")
+});
```

Comment:
There's a problem with this: `Lazy` is init-once. As far as I remember, that breaks the `Sighup` resetting of the rest of these. This means it is possible to get a highly invalid configuration going with PL/Rust after a sighup if this code isn't run again, which requires this value to be mutated (rerunning this code).

Reply:
Yea that's a good point. I wanted to avoid reading a file/parsing it every time a function is created but that seems small in the grand scheme of things

Reply:
https://docs.rs/once_cell/latest/once_cell/sync/struct.Lazy.html#method.get_mut ?

Reply:
Thanks! Thinking about this more, there seems to be two? cases where the file contents can be stale:

1. SIGHUP points the config to a different file
2. The contents of the file has been updated, but the file remains the same

For 1) you could just cache the previous filename and check if the GUC has changed to decide when to re-read the file, but for 2) if the file contents has been updated the only way to tackle that would be to re-read it every time? In which case maybe we should just read it every time if it's a case we want to handle.

Reply:
There are almost 100,000 crates on crates.io, so... I am not against reading it every time in principle, but I am not sure if it is actually a good idea to simply blindly open it each time.